Amethyst/Managers/ScreenManager.swift

```diff
@@ -93,8 +93,12 @@ final class ScreenManager<Window: WindowType>: NSObject {
     }
 
     func setNeedsReflowWithWindowChange(_ windowChange: Change<Window>) {
-        if case let .add(window) = windowChange {
+        switch windowChange {
+        case let .focusChanged(window):
             lastFocusedWindow = window
+        case .remove:
```

Comment:
You should take the `window` and make sure it's actually the `lastFocusedWindow` before clearing it.